### PR TITLE
initial commit

### PR DESCRIPTION
**Pull request by sbryzak**
Initial project structure

### CODE REVIEW

File: gormapplication/application.go [hunk 1]
```diff
@@ -0,0 +1,213 @@
+package gormapplication
+
+import (
+	"fmt"
+	"strconv"
+
+	//permissionrepo "github.com/codeready-toolchain/sandbox-auth/pkg/authorization/permission/repository"
+	//tokenrepo "github.com/codeready-toolchain/sandbox-auth/pkg/authorization/token/repository"
+
```
Comment (alexeykazakov) [inline gormapplication/application.go]:
```suggestion
```

File: gormapplication/application.go [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+	factorymanager "github.com/codeready-toolchain/sandbox-auth/pkg/application/factory/manager"
+	servicecontext "github.com/codeready-toolchain/sandbox-auth/pkg/application/service/context"
+	"github.com/codeready-toolchain/sandbox-auth/pkg/application/service/factory"
+	"github.com/codeready-toolchain/sandbox-auth/pkg/application/transaction"
+	accountrepo "github.com/codeready-toolchain/sandbox-auth/pkg/authentication/account/repository"
+	//providerrepo "github.com/codeready-toolchain/sandbox-auth/pkg/authentication/provider/repository"
```
Comment (alexeykazakov) [inline gormapplication/application.go]:
```suggestion
```

File: pkg/authorization/token/manager/token_manager.go [hunk 3]
```diff
@@ -0,0 +1,1139 @@
...
+	ExtractIdentityID(ctx context.Context) (uuid.UUID, error)
+	ExtractSubject(ctx context.Context) (string, error)
+	ParseToken(ctx context.Context, tokenString string) (*TokenClaims, error)
+	ParseTokenWithMapClaims(ctx context.Context, tokenString string) (jwt.MapClaims, error)
+	PublicKey(keyID string) *rsa.PublicKey
+	JSONWebKeys() tokenpkg.JSONKeys
+	PemKeys() tokenpkg.JSONKeys
+	KeyFunction(context.Context) jwt.Keyfunc
+	AuthServiceAccountToken() string
+	GenerateServiceAccountToken(saID string, saName string) (string, error)
```
Comment (alexeykazakov) [inline pkg/authorization/token/manager/token_manager.go]: Let's drop the SA token support and focus on login and identity federation only.

File: pkg/configuration/configuration.go [hunk 4]
```diff
@@ -0,0 +1,639 @@
...
+			"settings": allSettings,
+			"err":      err,
+		}).Panicln("Failed to marshall config to string")
+	}
+	return fmt.Sprintf("%s\n", y)
+}
+
+// NewConfigurationData creates a configuration reader object using configurable configuration file paths
+func NewConfigurationData(mainConfigFile string) (*ConfigurationData, error) {
+	c := &ConfigurationData{
```
Comment (alexeykazakov) [inline pkg/configuration/configuration.go]: Let's simplify the configuration. Let's use the env vars only and drop config files or command line params. We never used those in the auth service.

**Comment by alexeykazakov** [comment]
It doesn't look like just a project structure to be honest... Can we split it to:
- the bare minimum of the project structure
- PRs with individual core components such as: Configuration, TokenManager, etc.

**Comment by alexeykazakov** [comment]
And there is no tests at all.

**Comment by xcoulon** [comment]
also, do we need all the wrapper/factory boiler-plate code? It seems like you're bringing a large part of the "fabric8-auth framework" which was quite complex to understand and to use, IIRC. 
I wished we could have something simpler or maybe use an existing framework such as https://gobuffalo.io/ for example.

**Review by alexeykazakov — APPROVED** [review]
Please make sure we have a good test coverage for everything we can test. That's OK if you add tests and CI in the following PRs.

**Pull request merged**